app/static/src/app/store/appState/actions.ts

```diff
@@ -29,5 +44,28 @@ export const appStateActions: ActionTree<AppState, AppState> = {
                 await dispatch(`model/${ModelAction.DefaultModel}`);
             }
         }
+    },
+
+    async [AppStateAction.QueueStateUpload](context) {
+        const { state, commit } = context;
+        const isFitting = () => { return (state.appType === AppType.Fit) && ((state as FitState).modelFit.fitting); };
+
+        // Do not queue uploads while fitting is true - we'll upload when fit finishes
+        if (!isFitting()) {
+            // remove any existing queued upload, as this request should supersede it
+            commit(AppStateMutation.ClearQueuedStateUpload);
+
+            const queuedId: number = window.setInterval(() => {
+                // wait for any ongoing uploads to finish before starting a new one
+                // and do not actually upload while fitting is true
+                if (!state.stateUploadInProgress && !isFitting()) {
+                    commit(AppStateMutation.ClearQueuedStateUpload);
+                    immediateUploadState(context);
+                }
+            }, 2000);
```

Comment:
How hard would it be to make this a configuration parameter? If we find that the load is too high, or it's a bad experience to have it too long, it might be nice to be able to change it easily

Reply:
Good idea